src/Analysis/Ast/Impl/Analyzer/PythonAnalyzer.cs

```diff
@@ -162,6 +163,24 @@ public sealed class PythonAnalyzer : IPythonAnalyzer, IDisposable {
             }
         }
 
+        public void LintModule(IPythonModule module) {
+            if (module.ModuleType != ModuleType.User) {
+                return;
+            }
+
+            var optionsProvider = _services.GetService<IAnalysisOptionsProvider>();
+            if (optionsProvider?.Options?.LintingEnabled != false) {
+                ((ExpressionEval)module.Analysis.ExpressionEvaluator).RemoveDiagnostics(DiagnosticSource.Linter);
+                var linter = new LinterAggregator();
+                linter.Lint(module.Analysis, _services);
+
+                var afterLinter = module.Analysis.Diagnostics.ToArray();
```

Comment:
If linter has its own `DiagnosticSource`, why do we care about `Analysis.Diagnostics`?

Reply:
Currently it adds to the analysis diagnostics. However... come to think of it - maybe it should be separate?